src/sentry/scripts/similarity/index.lua

```diff
@@ -129,6 +136,31 @@ local function build_argument_parser(fields)
     end
 end
 
+local function build_variadic_argument_parser(fields, validator)
+    if validator == nil then
+        validator = identity
+    end
+    local parser = build_argument_parser(fields)
+    return function (arguments, offset)
+        if offset == nil then
+            offset = 0
+        end
+        if #arguments % #fields ~= 0 then
+            -- TODO: make this error less crummy
+            error('invalid number of arguments')
+        end
+        local results = {}
+        for i = 1, #arguments, #fields do
+            local value, _ = parser(table.slice(arguments, i, i + #fields - 1), i)
```

Comment:
Going to kick this one down the road for a bit because I don't think it's going to make a significant performance improvement here, but I'd like to move a lot of the functional stuff to iterators in the future for improved memory utilization so I will probably tackle it then.